### DISCUSSION TITLE
Will using a lot of small Livewire components result in bad performce?

### DISCUSSION THREAD

**Discussion by legodion** [livewire/livewire#4827] [1. Help]
I once read that Livewire is meant to be used for big components.

But can it be used for a lot of small components?

Would there be big performance issues if I used Livewire this way?

**Comment by mikebronner** [comment]
We have run into performance issues with many Livewire components if they hold complex items in public class properties, like models and collections. To work around this we are implementing deferred loading (https://laravel-livewire.com/docs/2.x/defer-loading) on all our components now, and performance is great. Unfortunately this puts all the performance onus on the browser now, and if the visitor has a slow computer, it can take significantly longer than someone with a performant computer.
